nexus/db-queries/src/db/datastore/affinity.rs

```diff
@@ -341,78 +349,271 @@ impl DataStore {
         opctx: &OpContext,
         authz_affinity_group: &authz::AffinityGroup,
         pagparams: &PaginatedBy<'_>,
-    ) -> ListResultVec<AffinityGroupInstanceMembership> {
+    ) -> ListResultVec<external::AffinityGroupMember> {
         opctx.authorize(authz::Action::Read, authz_affinity_group).await?;
 
-        use db::schema::affinity_group_instance_membership::dsl;
+        let mut query = QueryBuilder::new()
+            .sql(
+                "
+                SELECT * FROM (
+                SELECT instance.id as id,
+                    instance.name as name,
+                    instance.state,
+                    instance.migration_id,
+                    vmm.state
+                FROM affinity_group_instance_membership
+                INNER JOIN instance
+                ON instance.id = affinity_group_instance_membership.instance_id
+                LEFT JOIN vmm
+                ON instance.active_propolis_id = vmm.id
+                WHERE
+                    instance.time_deleted IS NULL AND
+                    vmm.time_deleted IS NULL AND
+                    group_id = ",
+            )
+            .param()
+            .bind::<diesel::sql_types::Uuid, _>(authz_affinity_group.id())
+            .sql(") ");
+
+        let (direction, limit) = match pagparams {
```

Comment:
I'm doing pagination manually in this PR, and I'm refactoring it in https://github.com/oxidecomputer/omicron/pull/7851.

My goal was to have sufficient test coverage here that the refactor is basically "just clean up"